### PR TITLE
Upgrade TensorRT to v7.2.2

### PR DESCRIPTION
**Pull request by stevenlix**
1. Upgrade TensorRT docker container to 20.12 with Ubuntu20.04, CUDA11 and CuDNN8
2. Update dependencies for Ubuntu20.04
3. Update TensorRT Linux GPU CI pipelines
4. Update TensorRT Windows GPU CI pipelines
    WIP for new TRT package installation on CI machines

### CODE REVIEW

**Comment by gwang-msft** [comment]
You need to update the https://github.com/microsoft/onnxruntime/blob/master/cgmanifests/submodules/cgmanifest.json
See https://github.com/microsoft/onnxruntime/blob/master/cgmanifests/README.md

File: .gitmodules [hunk 1]
```diff
@@ -56,9 +56,9 @@
 [submodule "cmake/external/SafeInt/safeint"]
 	path = cmake/external/SafeInt/safeint
 	url = https://github.com/dcleblanc/SafeInt.git
-[submodule "cmake/external/onnx-tensorrt"]
-	path = cmake/external/onnx-tensorrt
-	url = https://github.com/onnx/onnx-tensorrt.git
 [submodule "cmake/external/optional-lite"]
 	path = cmake/external/optional-lite
 	url = https://github.com/martinmoene/optional-lite.git
+[submodule "cmake/external/onnx-tensorrt"]
```
Comment (gwang-msft) [inline .gitmodules:62]: You need to update the https://github.com/microsoft/onnxruntime/blob/master/cgmanifests/submodules/cgmanifest.json
See https://github.com/microsoft/onnxruntime/blob/master/cgmanifests/README.md

**Comment by rafale77** [comment]
Hoping to get this in and wish you could provide pre-compiled binary with trt like the gpu version. Tried compiling the branch myself but ran out of resources because of this project's approach of not using installed dependencies on the host system but instead wanting to download and rebuild new ones.

**Comment by jywu-msft** [comment]
merge master to resolve the windows CI issues.

**Review by jywu-msft — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**